docs/quickstart_insights_core.rst

```diff
@@ -37,7 +36,8 @@ file associated with the insights-core project.
 .. HINT::
    You might also need to install ``gcc`` to be able to build some python modules,
    ``unzip`` to be able to run `pytest` on the ``insights-core`` repo,
-   and ``pandoc`` to build Insights Core documentation.
+   ``pandoc`` to build Insights Core documentation, and ``platform-python-devel`` 
+   and ``openssl-devel`` to build ``rhsm``.
```

Comment:
@ahitacat Since this is for the client only could you please remove the updates to the insights core quickstart doc. You may want to try and update the developer setup doc in the insights-client repo. https://github.com/RedHatInsights/insights-client

Reply:
@ryan-blakley This is actually what you need to run code in _insights-core_. _insights-client_ only warns you on make if you are missing the Python _rhsm_ module, but does not use it itself.

Reply:
@Glutexo I've never had to build rhsm to run code in insights-core. If it's required for the client code development, I'd recommend creating a section in the document that specifically mentions for client code development you may need to install x additionally. That way it's distinguished from what users viewing this doc need for rule development.

Reply:
E.g. the _auto_config_ module here in _insights-core_ uses the _rhsm_ module. https://github.com/RedHatInsights/insights-core/blob/e6b64c3c02752a737e61123f9167903cc058c5dd/insights/client/auto_config.py#L102 It may not be necessary for **rule** development, but it is for development of code in the _insights-core_ repository.

This change is in a page called “Quickstart Insights Development” with “Rule Development Setup” being just a subsection. The green “Hint” box even mentions (for _pytest_ in this case) the “insights-core repo” that includes the client code (and tests of it) as well. Adding a section sounds like a good idea. Thanks! 👍🏻

Reply:
We talked a bit about this with @ahitacat and it may be ok enough to mention in the _Hint_ box something like

> For the Client development, you may need to install `python-devel` and `openssl-devel`. This is not necessary for Rule only development.

However we should make sure first that adding `rhsm` as a PyPI dependency is the right way. If we find a way to directly use `rhsm` installed from an RPM, this is all void.

Reply:
After talking with Chris Snyder, we decided we’ll go the PyPI way. It’s not ready yet though and the version of `rhsm` in _PyPI_ is obsolete. We’re not going to install it and thus the `-devel` libraries are not necessary.